scala/scala.bzl

```diff
@@ -656,3 +656,52 @@ def scala_export_to_java(name, exports, runtime_deps):
     jars = jars,
     runtime_deps = ["@scala//:lib/scala-library.jar"] + runtime_deps
   )
+
+# This auto-generates a test suite based on the passed set of targets
+# we will add a root test_suite with the name of the passed name
+def scala_test_suite(name, srcs = [], deps = [], runtime_deps = [], data = [], resources = [],
+                     scalacopts = [], jvm_flags = [], visibility = None, size = None):
+    ts = []
+    for test_file in srcs:
+        n = "suite_test_%s_%s_test" % (name, test_file)
+        scala_test(n, srcs = [test_file], deps = deps, runtime_deps = runtime_deps, resources=resources, scalacopts=scalacopts, jvm_flags=jvm_flags, visibility=visibility, size=size)
+        ts.append(n)
+    native.test_suite(name = name, tests = ts)
+
+def scala_library_suite(name,
+                        srcs = [],
+                        deps = [],
+                        exports = [],
+                        plugins = [],
+                        runtime_deps = [],
+                        data = [],
+                        resources = [],
+                        resource_strip_prefix = "",
+                        scalacopts = [],
+                        javacopts = [],
+                        jvm_flags = [],
+                        print_compile_time = False,
+                        visibility = None
+                        ):
+    ts = []
+    for src_file in srcs:
+        n = "%s_lib_%s" % (name, src_file)
+        scala_library(name = n,
+                      srcs = [src_file],
+                      deps = deps,
+                      plugins = plugins,
+                      runtime_deps = runtime_deps,
+                      data = data,
+                      resources=resources,
+                      resource_strip_prefix = resource_strip_prefix,
+                      scalacopts = scalacopts,
+                      javacopts = javacopts,
+                      jvm_flags = jvm_flags,
+                      print_compile_time = print_compile_time,
+                      visibility=visibility,
```

Comment:
should we expose the inner targets, or should we make them private, and only use the visibility on the item below?